### DISCUSSION TITLE
Cannot retrieve PLAYER SCORE over BCP interface to external RPI DMD

### DISCUSSION THREAD

**Discussion by TheSilverSquadron** [missionpinball/missionpinball#131] [Q&A]
Our goal is to build a PC running MPF with an external DMD connected to a RPI.
In order to show the players SCORES, we need to retrieve this information from the BCP interface events content.
I use the following simple python code to 'monitor' the PCB events on localhost port 5050 for now,
I use the 'DEMO-MAN' example game and start the game (mpf -X) . Then run python script below where
i start the game using switch 's_start' and add points with 's_left_slingshot'.
The issue is that I ONLY see the 'trigger' event every second, but not any 'score' event.
Any help would be very much appreciated...
Kind regards,

Floor Keukens

import socket
import time
with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
    s.bind(("localhost" , 5050 ))
    s.listen()
    conn, addr = s.accept()
    with conn:
        data = conn.recv(1024)
        print(data) 
        conn.sendall(b'hello?version=1.1\n')
        time.sleep(1)
        conn.sendall(b'reset_complete\n')

        data = conn.recv(1024)
        print(data) 
        time.sleep(1)
        print('Start Demolition Man')
        conn.sendall(b'switch?name=s_start&state=1\n')
        conn.sendall(b'switch?name=s_start&state=0\n')

        while True:
            time.sleep(0.1)
            print('Get response from MPF')   
            data = conn.recv(1024)
            print(data)            
            time.sleep(1)
            conn.sendall(b'switch?name=s_left_slingshot&state=1\n')
            conn.sendall(b'switch?name=s_left_slingshot&state=0\n')

**Comment by avanwinkle** [comment]
To subscribe to player variable events, send the command `monitor_start?category=player_vars` from your BCP client and MPF will register it for any player variable updates, including score. 

Other "category" values you may be interested in are `core_events`, `service_events`, `modes`, and `machine_vars`
